paperqa/settings.py

```diff
@@ -572,6 +573,26 @@ class Settings(BaseSettings):
         frozen=True,
     )
 
+    # imported here to avoid circular ref
+    if TYPE_CHECKING:
+        from .agents.env import EnvironmentState
+    callbacks: Mapping[str, Callable[["EnvironmentState"], Any]] = Field(
```

Comment:
Pydantic `Field` descriptors only make sense within a `BaseModel`. Let's not use that here

Reply:
This is part of `Settings` is a class of `BaseSettings` which is a class of `BaseModel`
It makes sense here, no?

Reply:
This is within `Settings`, a class within `BaseSettings`, a class within `BaseModel`.
I think it makes sense here, no?

Reply:
I got thrown off by the `TYPE_CHECKING` not being at module level, resolving this one